src/content/Sync.js

```diff
@@ -310,40 +313,66 @@ gContactSync.Sync = {
   },
   /**
    * Sends an HTTP Request to Google for a feed of all the user's contacts.
-   * Calls gContactSync.Sync.sync with the response if successful or
+   * Calls gContactSync.Sync.sync2 with the contact entry array if successful or
    * gContactSync.Sync.syncNextUser on errors.
+   *
+   * @param aStartIndex {string} The index to start the request from.
    */
-  getContacts: function Sync_getContacts() {
+  getContacts: function Sync_getContacts(aStartIndex) {
     gContactSync.LOGGER.LOG("***Beginning Contact Synchronization***");
     var httpReq;
     if (gContactSync.Sync.mContactsUrl) {
       httpReq = new gContactSync.GHttpRequest("getFromGroup",
                                                   gContactSync.Sync.mCurrentAuthToken,
                                                   null,
                                                   null,
-                                                  gContactSync.Sync.mContactsUrl);
+                                                  gContactSync.Sync.mContactsUrl,
+                                                  aStartIndex);
     }
     else {
       httpReq = new gContactSync.GHttpRequest("getAll",
                                                   gContactSync.Sync.mCurrentAuthToken,
                                                   null,
-                                                  null);
+                                                  null,
+                                                  null,
+                                                  aStartIndex);
     }
     httpReq.mOnSuccess = function getContactsSuccess(httpReq) {
-      // gContactSync.serializeFromText does not do anything if verbose
-      // logging is disabled so the serialization won't waste time
-      var backup      = gContactSync.Sync.mBackup,
-          firstBackup = gContactSync.Sync.mFirstBackup,
-          feed        = gContactSync.serializeFromText(httpReq.responseText,
-                                                           backup);
-      gContactSync.LOGGER.VERBOSE_LOG(feed);
-      if (backup) {
-        gContactSync.gdata.backupFeed(feed,
-                                          gContactSync.Sync.mCurrentUsername,
-                                          (firstBackup ? "init_" : ""),
-                                          ".xml");
+      // Update the contact entries
+      var newContactEntries = Array.prototype.slice.call(httpReq.responseXML.getElementsByTagName('entry'), 0);
```

Comment:
Could you instead use
      var newContactEntries = Array.from(httpReq.responseXML.getElementsByTagName('entry'));

Reply:
Nevermind, I'm going to make a follow-up commit and will address this there.